public/app/features/plugins/extensions/usePluginComponents.tsx

```diff
@@ -28,47 +26,17 @@ export function usePluginComponents<Props extends object = {}>({
   const { isLoading: isLoadingAppPlugins } = useLoadAppPlugins(getExtensionPointPluginDependencies(extensionPointId));
 
   return useMemo(() => {
-    const isInsidePlugin = Boolean(pluginContext);
-    const isCoreGrafanaPlugin = pluginContext?.meta.module.startsWith('core:') ?? false;
-    const components: Array<ComponentTypeWithExtensionMeta<Props>> = [];
-    const extensionsByPlugin: Record<string, number> = {};
-    const pluginId = pluginContext?.meta.id ?? '';
-    const pointLog = log.child({
-      pluginId,
-      extensionPointId,
-    });
-
-    // Don't show extensions if the extension-point id is invalid in DEV mode
-    if (
-      isGrafanaDevMode() &&
-      !isExtensionPointIdValid({ extensionPointId, pluginId, isInsidePlugin, isCoreGrafanaPlugin, log: pointLog })
-    ) {
-      return {
-        isLoading: false,
-        components: [],
-      };
-    }
+    const { result } = getExtensionValidationResults({ extensionPointId, pluginContext, isLoadingAppPlugins });
 
-    // Don't show extensions if the extension-point misses meta info (plugin.json) in DEV mode
-    if (
-      isGrafanaDevMode() &&
-      !isCoreGrafanaPlugin &&
-      pluginContext &&
-      isExtensionPointMetaInfoMissing(extensionPointId, pluginContext)
-    ) {
-      pointLog.error(errors.EXTENSION_POINT_META_INFO_MISSING);
+    if (result) {
       return {
-        isLoading: false,
-        components: [],
+        isLoading: result.isLoading,
+        components: result.results,
```

Comment:
Hmm, I guess if there is a validation issue, then `result.results` cannot be anything else than an empty array, right? (Assuming we would like to return an empty list of extensions in case there are validation issues.)

What do you think, would it make sense to wire that in directly instead to simplify the logic? I am bringing it up because right now by looking at it I assumed that there is probably a case when it has actual results.

 ```suggestion
        components: [],
```

Reply:
Fixed 👍